src/System.Private.CoreLib/src/System/AppContext.Unix.cs

```diff
@@ -17,7 +19,34 @@ public static string BaseDirectory
         {
```

Comment:
We may want to cache the value to match the performance characteristic that people expect. It is cheap API on other runtimes because all other runtimes have some sort of cache for it.